tests/app_builder/with_wasm.rs

```diff
@@ -0,0 +1,173 @@
+use anyhow::Result as AnyResult;
+use cosmwasm_std::{
+    Addr, Api, Binary, BlockInfo, Deps, DepsMut, Empty, Env, MessageInfo, Querier, Record,
+    Response, StdError, Storage, SubMsg, WasmMsg, WasmQuery,
+};
+use cw_multi_test::{
+    AppBuilder, AppResponse, Contract, ContractData, ContractWrapper, CosmosRouter, Module, Wasm,
+};
+use schemars::JsonSchema;
+use std::fmt::Debug;
+use std::marker::PhantomData;
+
+mod contracts {
+    use super::*;
+
+    pub mod caller {
+        use super::*;
+
+        fn instantiate(
+            _deps: DepsMut,
+            _env: Env,
+            _info: MessageInfo,
+            _msg: Empty,
+        ) -> Result<Response, StdError> {
+            Ok(Response::default())
+        }
+
+        fn execute(
+            _deps: DepsMut,
+            _env: Env,
+            _info: MessageInfo,
+            msg: WasmMsg,
+        ) -> Result<Response, StdError> {
+            let message = SubMsg::new(msg);
+
+            Ok(Response::new().add_submessage(message))
+        }
+
+        fn query(_deps: Deps, _env: Env, _msg: Empty) -> Result<Binary, StdError> {
+            Err(StdError::generic_err(
+                "query not implemented for the `caller` contract",
+            ))
+        }
+
+        pub fn contract<C>() -> Box<dyn Contract<C>>
+        where
+            C: Clone + Debug + PartialEq + JsonSchema + 'static,
+        {
+            let contract = ContractWrapper::new_with_empty(execute, instantiate, query);
+            Box::new(contract)
+        }
+    }
+}
+
+#[test]
+fn building_app_with_custom_wasm_should_work() {
+    struct MyWasm<ExecT, QueryT>(PhantomData<(ExecT, QueryT)>);
+
+    impl<ExecT, QueryT> Default for MyWasm<ExecT, QueryT> {
+        fn default() -> Self {
+            Self(PhantomData)
+        }
+    }
+
+    impl<Exec, Query> Module for MyWasm<Exec, Query>
+    where
+        Exec: Debug,
+        Query: Debug,
+    {
+        type ExecT = Exec;
+        type QueryT = Query;
+        type SudoT = Empty;
```

Comment:
The question is, if it would be handy for users in the future to have something like FailingModule pre-implemented called for example `WasmModule`, so there would be no need to implement `Module`, or to provide `WasmModule` as a trait that implements defaults.

Reply:
I agree, I have removed the `C: Module` bound.